packages/editor/src/components/collab-sidebar/comments.js

```diff
@@ -0,0 +1,428 @@
+									className="editor-collab-sidebar-panel__child-thread"
+									id={ reply.id }
+									spacing="2"
+								>
+									<CommentBoard
+										thread={ reply }
+										parentThread={ thread }
+									/>
+								</VStack>
+							) ) }
+					</VStack>
+				) ) }
+		</>
+	);
+}
+
+/**
+ * EditComment component.
+ *
+ * @param {Object}   props          - The component props.
+ * @param {Function} props.onSubmit - The function to call when updating the comment.
+ * @param {Function} props.onCancel - The function to call when canceling the comment update.
+ * @param {Object}   props.thread   - The comment thread object.
+ * @return {JSX.Element} The CommentForm component.
+ */
+function CommentForm( { onSubmit, onCancel, thread } ) {
+	const [ inputComment, setInputComment ] = useState(
+		thread?.content?.raw ?? ''
+	);
+
+	return (
+		<>
+			<TextareaControl
+				__nextHasNoMarginBottom
+				className="editor-collab-sidebar-panel__comment-field-textarea"
+				value={ inputComment ?? '' }
+				onChange={ setInputComment }
+			/>
+			<VStack alignment="left" spacing="3" justify="flex-start">
+				<HStack alignment="left" spacing="3" justify="flex-start">
+					<Button
+						__next40pxDefaultSize
+						accessibleWhenDisabled
+						variant="primary"
+						onClick={ () => onSubmit( inputComment ) }
+						size="compact"
+						disabled={
+							0 === sanitizeCommentString( inputComment ).length
+						}
+					>
+						{ thread
+							? _x( 'Update', 'verb' )
+							: _x( 'Reply', 'Add reply comment' ) }
+					</Button>
+					<Button
+						__next40pxDefaultSize
+						onClick={ onCancel }
+						size="compact"
+					>
+						{ _x( 'Cancel', 'Cancel comment edit' ) }
+					</Button>
+				</HStack>
+			</VStack>
+		</>
+	);
+}
+
+/**
+ * Renders a confirmation notice component.
+ *
+ * @param {Object}   props                - The component props.
+ * @param {string}   props.confirmMessage - The confirmation message to display. Defaults to 'Are you sure?' if not provided.
+ * @param {Function} props.confirmAction  - The action to perform when the confirm button is clicked.
+ * @param {Function} props.discardAction  - The action to perform when the discard button is clicked.
+ * @return {JSX.Element} The confirmation notice component.
+ */
+function ConfirmNotice( { confirmMessage, confirmAction, discardAction } ) {
+	return (
+		<VStack
+			// translators: title for the confirmation overlay
+			title={ __( 'Confirm' ) }
+			className="editor-collab-sidebar-panel__thread-overlay"
+			spacing="0"
+			justify="space-between"
+		>
+			<p>
+				{ confirmMessage ??
+					// translators: message displayed when confirming an action
+					__( 'Are you sure?' ) }
+			</p>
+			<HStack>
+				<Button
+					__next40pxDefaultSize
+					variant="primary"
+					onClick={ confirmAction }
+					size="compact"
+				>
+					{ __( 'Yes' ) }
+				</Button>
+				<Button
+					__next40pxDefaultSize
+					onClick={ discardAction }
+					size="compact"
```

Comment:
I see we're specifying both `size="compact"` and `__next40pxDefaultSize`. Let's pick one and go with it.